src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -337,8 +338,22 @@ bool DirectoryService::VerifyPoWOrdering(const DequeOfShard& shards) {
       if (it == sortedPoWSolns.cend()) {
         LOG_GENERAL(WARNING, "Failed to find key in the PoW ordering "
                                  << toFind << " " << sortedPoWSolns.size());
-        ret = false;
-        break;
+
+        LOG_GENERAL(INFO,
+                    "Checking for the key and PoW in the announcement...");
+
+        auto itLeaderMap = allPoWsFromTheLeader.find(toFind);
+        if (itLeaderMap != allPoWsFromTheLeader.end()) {
+          LOG_GENERAL(INFO,
```

Comment:
Recommend to use `WARNING` instead of `INFO`

Reply:
Can be done in next PR